src/main/java/gregtech/integration/jei/basic/GTFluidVeinCategory.java

```diff
@@ -0,0 +1,180 @@
+        return gtFluidVeinInfo;
+    }
+
+    @Override
+    public void drawExtras(@Nonnull Minecraft minecraft) {
+
+        GTStringUtils.drawCenteredStringWithCutoff(veinName, minecraft.fontRenderer, 176);
+
+        this.slot.draw(minecraft, SLOT_CENTER - 1, 18);
+
+        // Vein Weight information
+        String veinWeight = I18n.format("gregtech.jei.fluid.vein_weight") + " " + weight;
+        weightLength = minecraft.fontRenderer.getStringWidth(veinWeight);
+        minecraft.fontRenderer.drawString(veinWeight, textStartX, startPosY, 0x111111);
+
+        // Vein Minimum Yield information
+        String veinMinYield = I18n.format("gregtech.jei.fluid.min_yield") + " " + yields[0];
+        minYieldLength = minecraft.fontRenderer.getStringWidth(veinMinYield);
+        minecraft.fontRenderer.drawString(veinMinYield, textStartX, startPosY + FONT_HEIGHT + 1, 0x111111);
+
+        // Vein Maximum Yield information
+        String veinMaxYield = I18n.format("gregtech.jei.fluid.max_yield") + " " + yields[1];
+        maxYieldLength = minecraft.fontRenderer.getStringWidth(veinMaxYield);
+        minecraft.fontRenderer.drawString(veinMaxYield, textStartX, startPosY + 2 * FONT_HEIGHT + 1, 0x111111);
+
+        // Vein Depletion Chance information
+        String veinDepletionChance = I18n.format("gregtech.jei.fluid.depletion_chance") + " " + depletionChance;
+        depletionChanceLength = minecraft.fontRenderer.getStringWidth(veinDepletionChance);
+        minecraft.fontRenderer.drawString(veinDepletionChance, textStartX, startPosY + 3 * FONT_HEIGHT + 1, 0x111111);
+
+        // Vein Depletion Amount information
+        String veinDepletionAmount = I18n.format("gregtech.jei.fluid.depletion_amount") + " " + depletionAmount;
+        depletionAmountLength = minecraft.fontRenderer.getStringWidth(veinDepletionAmount);
+        minecraft.fontRenderer.drawString(veinDepletionAmount, textStartX, startPosY + 4 * FONT_HEIGHT + 1, 0x111111);
+
+        // Vein Depleted Yield information
+        String veinDepletedYield = I18n.format("gregtech.jei.fluid.depleted_rate") + " " + depletedYield;
+        depletedYieldLength = minecraft.fontRenderer.getStringWidth(veinDepletedYield);
+        minecraft.fontRenderer.drawString(veinDepletedYield, textStartX, startPosY + 5 * FONT_HEIGHT + 1, 0x111111);
+
+        // Vein Dimensions information
+        String veinDimension = I18n.format("gregtech.jei.fluid.dimension") + " ";
+        int dimensionLength = minecraft.fontRenderer.getStringWidth(veinDimension);
+        minecraft.fontRenderer.drawString(veinDimension, textStartX, startPosY + 6 * FONT_HEIGHT + 1, 0x111111);
+
+        GTStringUtils.drawMultiLineCommaSeparatedDimensionList(namedDimensions, dimension.get(), startPosY + 6 * FONT_HEIGHT + 1, textStartX + dimensionLength);
+
+    }
+
+    @Nonnull
+    @Override
+    public List<String> getTooltipStrings(int mouseX, int mouseY) {
```

Comment:
This whole section is quite brutal. Maybe extracting check to own function like `mouseCursorInBounds` (maybe there is already some helper for it) which would just take mouse params and startX,Y and sizeX,Y of box.